autowarefoundation/autoware
# add tf generator #1238

## Status
**PRODUCTION / DEVELOPMENT**

## Description
This node tries to solve autowarefoundation/autoware_ai#138. I changed default launch file of tf in map.yaml to use new launch file.
The new lauch file runs the new node called map_tf_generator, calculates the coordinates of the Point Clouds map, and broadcasts /world to /map tf at 10 hertz.


## Todos


## Steps to Test or Reproduce
1. Checkout this branch

2. Start-up Runtime Manager
```
cd /path/to/Autoware/ros
./run
```

3. Publish Point Cloud Map in Map tab
4. Click TF button with no ref in Map tab
![screenshot from 2018-05-30 13-54-07](https://user-images.githubusercontent.com/38061530/40699733-110c9ffc-6411-11e8-9a97-433deb07a3b9.png)

5. You can see the Point Cloud map in RViz with Fixed Frame: world, and check with below.
```
rostopic echo /tf 
transforms: 
  - 
    header: 
      seq: 0
      stamp: 
        secs: 1527655944
        nsecs: 766463623
      frame_id: "world"
    child_frame_id: "map"
    transform: 
      translation: 
        x: 15705.4459683
        y: 85591.2237567
        z: -44.5270118697
      rotation: 
        x: 0.0
        y: 0.0
        z: 0.0
        w: 1.0
---
```
